### PR TITLE
[build] Build snap and push to latest/edge

### PR DESCRIPTION
**Pull request by arif-ali**
When a any MRs or commits are added to the main branch then automatically create a snap, and push to latest/edge channel

Signed-off-by: Arif Ali <arif.ali@canonical.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [ ] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by arif-ali** [comment]
This is based on the following 2 GH Actions

1. https://github.com/snapcore/action-build
1. https://github.com/snapcore/action-publish

I can provide the `STORE_LOGIN` details privately, if we want to go ahead with this.

This will will to get the latest sos as a snap, and we don't need to publish this manually

stable releases will still be done manually

File: .github/workflows/snap.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: snap
+on:
+  push:
+    tags:
+      - '*'
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```
Comment (TurboTurtle) [inline .github/workflows/snap.yaml:10]: Would this kick off a new build on any new PR looking to merge into main? If the intent is for only merged commits to create a build, shouldn't the `push` section above be sufficient?

**Review by TurboTurtle — COMMENTED** [review]
Ack to the end goal of this, just one question about the action configuration.

File: .github/workflows/snap.yaml (hunk 1 above)
Comment (arif-ali) [reply .github/workflows/snap.yaml]: ack, makes total sense :) totally missed that point

**Review by TurboTurtle — APPROVED** [review]

**Pull request merged**